### DISCUSSION TITLE
Lookup stopped working migrating to Dynamic Form Modal

### DISCUSSION THREAD

**Discussion by davidgarra** [mia-platform/community#366] [Q&A] [answered]
### Which section?

Backoffice

### Which is the version?

back-kit@1.4.7, microlc-backend@0.8.1, microlc-frontend@0.9.0

### Description

Hi, I've migrated a `Form Modal` to a `Dynamic Form Modal` and I've encountered an issue with a lookup that stopped working. Inspecting the network I noticed that the url is now wrong. With the `Form Modal` the contacted url for the lookup is `v2/users/`, whereas with the `Dynamic Form Modal` the url has changed to `lookup/userId/`.

This is the field:
```
        "userId": {
          "type": "string",
          "format": "lookup",
          "excludeFromSearch": false,
          "label": {
            "it": "Paziente/Genitore",
            "en": "Patient/Parent"
          },
          "formOptions": {
            "readOnlyOnUpdate": true
          },
          "lookupOptions": {
            "lookupValue": "_id",
            "lookupFields": [
              "firstName",
              "lastName",
              "fiscalCode"
            ],
            "lookupDelimiter": " ",
            "lookupDataSource": "v2/users",
            "lookupQueries": [
              {
                "property": "userGroup",
                "operator": "equal",
                "value": "customer"
              }
            ]
          }
        },
```

Looks like the `lookupDataSource` prop is being ignored... any suggestion?

**Comment by clabene** [comment, marked as answer]
Hi @davidgarra,
the Dynamic Form Modal component does not support lookup fields the way Form Modal does. Instead, it expects references to be resolved on the backend side. Values of lookup fields thus become objects with fields "value" and "label."
In particular, the Dynamic Form Modal component is designed to leverage [Writable Views](https://docs.mia-platform.eu/docs/runtime_suite/crud-service/writable_views), available in the CRUD Service.

More details are available [here](https://docs.mia-platform.eu/docs/microfrontend-composer/back-kit/components/dynamic_form_modal#working-with-views) and [here](https://docs.mia-platform.eu/docs/microfrontend-composer/back-kit/examples/lookups#migrating-to-writable-views-from-previous-lookups)

**Reply by davidgarra** [reply]
So for every `Form Modal` that is converted to a `Form Dynamic Modal` it is necessary to create writable views?

**Reply by davidgarra** [reply]
I've created a view as follows:
```
[
  {
    "$lookup": {
      "from": "users",
      "localField": "userId",
      "foreignField": "_id",
      "as": "user",
      "pipeline": [
        {
          "$project": {
            "_id": 0,
            "value": {
              "$toObjectId": "$_id"
            },
            "label": {
              "$toString": {
                "$concat": [
                  "$firstName",
                  " ",
                  "$lastName"
                ]
              }
            }
          }
        }
      ]
    }
  },
  {
    "$unwind": {
      "path": "$user",
      "preserveNullAndEmptyArrays": true
    }
  },
  {
    "$project": {
      "userId": 0
    }
  }
]
```

I can see the view result correctly, but going to the `/lookup/user` endpoint the system returns this error: 
```
{
  "statusCode": 500,
  "error": "Internal Server Error",
  "message": "Unknown field: __STATE__"
}
```

The view looks exactly like the one in the example to me...

The CRUD-service version is the 6.9.4

**Reply by davidgarra** [reply]
Just for anyone who will need this, I solved adding this to the internal pipeline in the view after the `_id` field:
```
            "__STATE__": {
              "$toString": "$__STATE__"
            },
```
Even though this is not written in the doc

**Reply by clabene** [reply]
You will need to create Writable Views using the Dynamic Form Modal component if lookup fields are included in the underlying collection. This is not necessary when the component interacts with data that do not include references to other collections.

As far as your difficulties with writing the view's pipeline, thanks for sharing the solution that worked for you!
In order to understand what went wrong when you applied the example from our doc, we would need further information about what steps were taken to create the view, and / or the schema of the starting collection.
If you are willing to share such info, we might be able to reproduce the error and add related information to our docs.

Thanks
